app/Helpers/database/game.php

```diff
@@ -998,6 +1012,29 @@ function GetPatchData(int $gameID, ?User $user, int $flag): array
         'Leaderboards' => [],
     ];
 
+    if ($clientSupportLevel !== ClientSupportLevel::Full) {
+        if ($game->achievements_published < 0) { // will never be true. change to > when ready
+            $gameData['Achievements'][] = [
+                'ID' => Achievement::CLIENT_WARNING_ID,
+                'MemAddr' => '1=1.300.', // pop after 5 seconds
+                'Title' => ($clientSupportLevel === ClientSupportLevel::Outdated) ?
+                    'Warning: Outdated Client (please update)' : 'Warning: Unknown Client',
+                'Description' => 'Hardcore unlocks cannot be earned using this client.',
```

Comment:
I'm wondering if it may be best to change the word "client" to "emulator" for these user-facing messages.

Players who are perhaps not as technically sophisticated may not know what "client" means (ie: they could interpret this as meaning some kind of option within the emulator needs to be adjusted).